sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.students.controller.getStudentProfile.e2e-spec.ts

```diff
@@ -64,9 +75,120 @@ describe("StudentInstitutionsController(e2e)-getStudentProfile", () => {
+
+    // Get any student user token.
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect((response) => expect(response.body.hasFulltimeAccess).toBe(true));
+  });
+
+  it("Should get the student profile with hasFulltimeAccess as false when the student was added to the beta users table with a future date and allowBetaUsersOnly is true.", async () => {
+    // Arrange
+    allowBetaUsersOnly(true);
+    const student = await saveFakeStudent(db.dataSource);
+    // Register the student as a beta user for full-time.
+    await db.betaUsersAuthorizations.save({
+      givenNames: student.user.firstName,
+      lastName: student.user.lastName,
+      enabledFrom: addDays(1),
+    });
+
+    // Mock user service to return the saved student.
+    await mockUserLoginInfo(appModule, student);
+
+    // Get any student user token.
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect((response) =>
+        expect(response.body.hasFulltimeAccess).toBe(false),
+      );
+  });
+
+  it("Should get the student profile with hasFulltimeAccess as false when the student was not added to the beta users table and allowBetaUsersOnly is true.", async () => {
+    // Arrange
+    allowBetaUsersOnly(true);
+    const student = await saveFakeStudent(db.dataSource);
+
+    // Mock user service to return the saved student.
+    await mockUserLoginInfo(appModule, student);
+
+    // Get any student user token.
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect((response) =>
+        expect(response.body.hasFulltimeAccess).toBe(false),
+      );
+  });
+
+  it("Should get the student profile with hasFulltimeAccess as true when the student was not added to the beta users table but allowBetaUsersOnly is false.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(db.dataSource);
+
+    // Mock user service to return the saved student.
+    await mockUserLoginInfo(appModule, student);
+
+    // Get any student user token.
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect((response) => expect(response.body.hasFulltimeAccess).toBe(true));
+  });
```

Comment:
This scenario is implicitly asserted in the first test in the suite. IMO this one can be removed. Just a suggestion and not a blocker to keep it.

Reply:
I will remove it.

Reply:
Removed.